sklearn/feature_selection/_rfe.py

```diff
@@ -786,11 +785,18 @@ def fit(self, X, y, groups=None):
 
         # reverse to stay consistent with before
         scores_rev = scores[:, ::-1]
-        self.cv_results_ = {}
-        self.cv_results_["mean_test_score"] = np.mean(scores_rev, axis=0)
-        self.cv_results_["std_test_score"] = np.std(scores_rev, axis=0)
-
-        for i in range(scores.shape[0]):
-            self.cv_results_[f"split{i}_test_score"] = scores_rev[i]
+        self.cv_results_ = {
+            "n_features": np.flip(
+                np.append(
+                    np.arange(
+                        start=n_features, stop=self.min_features_to_select, step=-step
+                    ),
+                    self.min_features_to_select,
+                )
+            ),
```

Comment:
I'd rather add this key last, in case some users rely on the order (they shouldn't but we don't want to break existing code).